src/globus_sdk/utils.py

```diff
@@ -44,20 +45,23 @@ def slash_join(a: str, b: str | None) -> str:
 
 
 def safe_strseq_iter(
-    value: t.Iterable[t.Any],
+    value: t.Iterable[t.Any] | str | uuid.UUID,
 ) -> t.Generator[str, None, None]:
     """
     Given an Iterable (typically of strings), produce an iterator over it of strings.
     This is a passthrough with two caveats:
     - if the value is a solitary string, yield only that value
-    - str value in the iterable which is not a string
+    - if the value is a solitary UUID, yield only that value (as a string)
+    - str values in the iterable which are not strings
 
     This helps handle cases where a string is passed to a function expecting an iterable
     of strings, as well as cases where an iterable of UUID objects is accepted for a
     list of IDs, or something similar.
     """
     if isinstance(value, str):
         yield value
+    elif isinstance(value, uuid.UUID):
+        yield str(value)
```

Comment:
Is this function unit tested, or is it implicitly tested by other code?

If unit tests exist for this (and I hope they do!) then I'd like to have them updated to confirm UUIDs are handled correctly.

Reply:
I think that -- and this is surprising to me -- we don't have dedicated unit tests for this, only the implicit runs of it via functional tests.
I'm adding a couple of quick tests now, so we'll get this included in the PR.